_posts/2017-05-19-build-your-own-static-vector-tile-pipeline.md

```diff
@@ -0,0 +1,764 @@
+pre-coloured data) too:
+
+* Map clients can access the vector data directly so have the ability to
+  re-colour or style the data dynamically themselves without needing to load more
+  data from the server (something that can't be done with raster tiles, which are
+  just images of the data)
+
+* Map clients can zoom in and out smoothly between the different zoom levels
+  because the data can be easily scaled while new data is loaded. This is known
+  as *overzooming* or *underzooming*.
+
+* Vector tiles often have smaller file sizes than raster tiles. For example
+  imagine you want to draw a line. In an image, you'd need to colour in each
+  pixel between the start and end point (perhaps 20 numbers), whereas as a
+  vector, you just specify numbers for the start and end point.
+
+These advantages of vector tiles all add up to:
+
+* A faster and smoother experience for the end user
+* Lower bandwidth costs for the hosting
+* The potential for building more advanced map applications that make use of
+  the data transmitted
+
+
+## Hosting
+
+Vector tiles are also fairly easy to host cheaply since they can be represented
+as a simple directory structure of files.
+
+Google Cloud Storage is one place to host the tiles so that they can be
+retrieved by a map client. The advantage over running your own server is that
+Google will look after keeping the files accessible so that you don't need
+hosting expertise or 24 hour monitoring and response to keep your tiles
+available. It is also a cheaper option than hosting directly with Mapbox.
+
+Modern browsers are able to read compressed files directly if they are
+compressed in gzip format and served with the following header:
+
+```
+Content-Encoding: gzip
+```
+
+In this post we'll assume you have a host that is capable of adding this header
+to HTTP responses so that we can store and transfer compressed tiles, rather
+than uncompressed ones which take up more space and take longer to transfer.
+
+
+## Getting Started
+
+To get started, why not download some shapefiles for London to experiment with from
+the [Ordnance Survey Open Local download
+page](https://www.ordnancesurvey.co.uk/opendatadownload/products.html#OPMPLC).
+
+You'll need to tick the box to the right of the `National Grid Reference
+squares` label, in the `Download` column, and then scroll down to select `TQ`
+from the list next to the grid squares diagram. Scroll to the very bottom of
+the page, click `Continue` and follow the instructions.
+
+The page mentions that it can take up to 2 hours to fulfil the request. In
+practice it is nearly instant so don't be put off.
+
+Repeat the process to download the `TR` grid square data too. (You can also
+request them both at the same time by selecting them both to start with).
+
+Note: Each download has a `data` directory containing different *shapefiles*
+representing the different types of data available. Confusingly, each shapefile
```

Comment:
This is a repeat of what you have already said above (about 4 shapefiles being confusing)